tests/spread/basic/task.yaml

```diff
@@ -0,0 +1,42 @@
+summary: Ensure that things worked
+
+execute: |
+  # for various utilities
+  . "$TESTSLIB/nested.sh"
+
+  # Start the nested UC20 VM
+  start_nested_core_vm_unit
+
+  # At this point we are able to SSH to the nested VM, so things probably worked
+  # but we should wait for snapd to finish seeding anyways, but first we need to
+  # wait for the snap command to exist
+  retry=400
+  wait=1
+  while ! execute_remote command -v snap; do
+      retry=$(( retry - 1 ))
+      if [ $retry -le 0 ]; then
+          echo "Timed out waiting for snap command to be available. Aborting!"
+          exit 1
+      fi
+      sleep "$wait"
+  done
+
+  # Now formally wait for seeding
+  execute_remote sudo snap wait system seed.loaded
+
+  # We should be in run mode
+  execute_remote cat /proc/cmdline | MATCH snapd_recovery_mode=run
+
+  # We should have a modeenv
+  execute_remote test -f /var/lib/snapd/modeenv
+
+  # Verify that the partitions we expect, are present
+  execute_remote lsblk -f -m -l | awk '{ print $3 }' | grep ubuntu-seed | xargs test
+  execute_remote lsblk -f -m -l | awk '{ print $3 }' | grep ubuntu-boot | xargs test
+  execute_remote lsblk -f -m -l | awk '{ print $3 }' | grep ubuntu-save | xargs test
+  execute_remote lsblk -f -m -l | awk '{ print $3 }' | grep ubuntu-data | xargs test
+  
+  # Verify system services are running
+  execute_remote systemctl -q is-system-running --wait
+  execute_remote systemctl -q is-active snapd
```

Comment:
An additional check you could add is to make sure that `snap change 1` (system initialization) has finished its tasks without errors.

Reply:
Another good point! I'll add that to the list of tests.